### ISSUE TITLE
Date getting parsed as Interval for TPCH data

### ISSUE THREAD

**Issue by snehakrishnamurthy** [UBOdin/mimir#272]
Date type fields in TPCH data gets parsed as Interval when using the tuplebundle compile mode for query execution where the date field is non deterministic.

**Comment by okennedy** [comment]
Odd.  Probably an issue with the regexps.  I'll look into it.

**Comment by mrb24** [comment]
I am not seeing this.  Sneha,  can you post a query that causes this.

**Comment by snehakrishnamurthy** [comment]
This is the exeption : mimir.algebra.DatePrimitive cannot be cast to mimir.algebra.IntervalPrimitive
 when a query with uncertain date values is executed.
Eg: select shipdate from lineitem_run_$i; where shipdate is uncertain

**Comment by snehakrishnamurthy** [comment]
The exception is thrown during the creation of missing value lens where date type is missing

```
 create lens b as select * from lineitem with missing_value('shipdate');
Oct 28, 2017 7:36:13 PM com.github.fommil.netlib.ARPACK <clinit>
WARNING: Failed to load implementation from: com.github.fommil.netlib.NativeSystemARPACK
Oct 28, 2017 7:36:13 PM com.github.fommil.jni.JniLoader liberalLoad
INFO: successfully loaded C:\Users\Sneha\AppData\Local\Temp\jniloader5562547521430287087netlib-native_ref-win-x86_64.dll
mimir> select * from sys_tables;
 TABLE_NAME               | SCHEMA_NAME
--------------------------+-------------
 'Nation'                 | 'BACKEND'
 'Part'                   | 'BACKEND'
 'Orders'                 | 'BACKEND'
 'LineItem'               | 'BACKEND'
 'Supplier'               | 'BACKEND'
 'Region'                 | 'BACKEND'
 'PartSupp'               | 'BACKEND'
 'Customer'               | 'BACKEND'
 'MIMIR_MODELS'           | 'BACKEND'
 'MIMIR_MODEL_OWNERS'     | 'BACKEND'
 'MIMIR_VIEWS'            | 'BACKEND'
 'MIMIR_ADAPTIVE_SCHEMAS' | 'BACKEND'
 'WORLDBits'              | 'BACKEND'
 'LINEITEM_RUN_1'         | 'MIMIR'
 'ORDERS_RUN_1'           | 'MIMIR'
 'B'                      | 'MIMIR'
QUERY took: 22 ms
mimir> select * from b limit 10;
java.lang.ClassCastException: mimir.algebra.DatePrimitive cannot be cast to mimir.algebra.IntervalPrimitive
        at mimir.exec.EvalInlined$$anonfun$compileForInterval$2.apply(EvalInlined.scala:288)
        at mimir.exec.EvalInlined$$anonfun$compileForInterval$2.apply(EvalInlined.scala:288)
        at mimir.exec.EvalInlined$$anonfun$compilePassthrough$3.apply(EvalInlined.scala:269)
        at mimir.exec.EvalInlined$$anonfun$compileConditional$1.apply(EvalInlined.scala:88)
        at mimir.exec.EvalInlined$$anonfun$checkNull$1.apply(EvalInlined.scala:21)
        at mimir.exec.EvalInlined$$anonfun$checkNull$1.apply(EvalInlined.scala:20)
        at mimir.exec.result.LazyRow.apply(LazyRow.scala:16)
        at mimir.exec.PrettyOutputFormat$$anonfun$print$4$$anonfun$apply$1.apply$mcVI$sp(OutputFormat.scala:76)
        at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:160)
        at mimir.exec.PrettyOutputFormat$$anonfun$print$4.apply(OutputFormat.scala:75)
        at mimir.exec.PrettyOutputFormat$$anonfun$print$4.apply(OutputFormat.scala:74)
        at scala.collection.Iterator$class.foreach(Iterator.scala:891)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
        at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
        at mimir.exec.result.ResultSeq.foreach(ResultSeq.scala:5)
        at mimir.exec.PrettyOutputFormat.print(OutputFormat.scala:74)
        at mimir.Mimir$$anonfun$handleQuery$1$$anonfun$apply$mcV$sp$1.apply(Mimir.scala:154)
        at mimir.Mimir$$anonfun$handleQuery$1$$anonfun$apply$mcV$sp$1.apply(Mimir.scala:154)
        at mimir.Database.query(Database.scala:127)
        at mimir.Database.query(Database.scala:168)
        at mimir.Mimir$$anonfun$handleQuery$1.apply$mcV$sp(Mimir.scala:154)
        at mimir.Mimir$$anonfun$handleQuery$1.apply(Mimir.scala:154)
        at mimir.Mimir$$anonfun$handleQuery$1.apply(Mimir.scala:154)
        at mimir.util.TimeUtils$.monitor(TimeUtils.scala:20)
        at mimir.Mimir$.handleQuery(Mimir.scala:153)
        at mimir.Mimir$.handleSelect(Mimir.scala:148)
        at mimir.Mimir$.eventLoop(Mimir.scala:112)
        at mimir.Mimir$.main(Mimir.scala:94)
        at mimir.Mimir.main(Mimir.scala)
```

**Comment by mrb24** [comment]
Mimir should not be loading this library:
com.github.fommil.netlib.NativeSystemARPACK
It should instead be loading:
com.github.fommil.netlib.F2jARPACK

Sneha,
Is your build.sbt up to date with master?
If so what platform are you running on?

**Comment by snehakrishnamurthy** [comment]
I was on a older build while testing this, will check it with the new build and get back

**Comment by snehakrishnamurthy** [comment]
I get this when i try to create a missing value lens with the new build.

```
mimir> create lens a as select * from lineitem with missing_value('shipdate');
17:06:25.032 [main] WARN  o.a.hadoop.util.NativeCodeLoader - Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
17:06:25.115 [main] ERROR org.apache.hadoop.util.Shell - Failed to locate the winutils binary in the hadoop binary path
java.io.IOException: Could not locate executable null\bin\winutils.exe in the Hadoop binaries.
        at org.apache.hadoop.util.Shell.getQualifiedBinPath(Shell.java:378)
        at org.apache.hadoop.util.Shell.getWinUtilsPath(Shell.java:393)
        at org.apache.hadoop.util.Shell.<clinit>(Shell.java:386)
        at org.apache.hadoop.util.StringUtils.<clinit>(StringUtils.java:79)
        at org.apache.hadoop.security.Groups.parseStaticMapping(Groups.java:116)
        at org.apache.hadoop.security.Groups.<init>(Groups.java:93)
        at org.apache.hadoop.security.Groups.<init>(Groups.java:73)
        at org.apache.hadoop.security.Groups.getUserToGroupsMappingService(Groups.java:293)
        at org.apache.hadoop.security.UserGroupInformation.initialize(UserGroupInformation.java:283)
        at org.apache.hadoop.security.UserGroupInformation.ensureInitialized(UserGroupInformation.java:260)
        at org.apache.hadoop.security.UserGroupInformation.loginUserFromSubject(UserGroupInformation.java:789)
        at org.apache.hadoop.security.UserGroupInformation.getLoginUser(UserGroupInformation.java:774)
        at org.apache.hadoop.security.UserGroupInformation.getCurrentUser(UserGroupInformation.java:647)
        at org.apache.spark.util.Utils$$anonfun$getCurrentUserName$1.apply(Utils.scala:2430)
        at org.apache.spark.util.Utils$$anonfun$getCurrentUserName$1.apply(Utils.scala:2430)
        at scala.Option.getOrElse(Option.scala:121)
        at org.apache.spark.util.Utils$.getCurrentUserName(Utils.scala:2430)
        at org.apache.spark.SparkContext.<init>(SparkContext.scala:295)
        at mimir.ml.spark.SparkML.getSparkSession(SparkML.scala:33)
        at mimir.ml.spark.SparkML.getSparkSqlContext(SparkML.scala:45)
        at mimir.ml.spark.SparkML.prepareData(SparkML.scala:78)
        at mimir.ml.spark.Classification$$anonfun$NaiveBayesMulticlassModel$1.apply(Classification.scala:143)
        at mimir.ml.spark.Classification$$anonfun$NaiveBayesMulticlassModel$1.apply(Classification.scala:142)
        at mimir.models.SimpleSparkClassifierModel$$anonfun$train$1.apply$mcV$sp(SparkClassifierModel.scala:87)
        at mimir.models.SimpleSparkClassifierModel$$anonfun$train$1.apply(SparkClassifierModel.scala:85)
        at mimir.models.SimpleSparkClassifierModel$$anonfun$train$1.apply(SparkClassifierModel.scala:85)
        at mimir.util.Timer$.monitor(Timer.scala:22)
        at mimir.models.SimpleSparkClassifierModel.train(SparkClassifierModel.scala:85)
        at mimir.models.SparkClassifierModel$$anonfun$train$2.apply(SparkClassifierModel.scala:37)
        at mimir.models.SparkClassifierModel$$anonfun$train$2.apply(SparkClassifierModel.scala:30)
        at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
        at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
        at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
        at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
        at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
        at scala.collection.AbstractTraversable.map(Traversable.scala:104)
        at mimir.models.SparkClassifierModel$.train(SparkClassifierModel.scala:30)
        at mimir.models.ModelRegistry$$anonfun$1.apply(ModelRegistry.scala:101)
        at mimir.models.ModelRegistry$$anonfun$1.apply(ModelRegistry.scala:101)
        at mimir.lenses.MissingValueLens$$anonfun$4.apply(MissingValueLens.scala:101)
        at mimir.lenses.MissingValueLens$$anonfun$4.apply(MissingValueLens.scala:95)
        at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
        at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
        at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
        at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
        at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
        at scala.collection.AbstractTraversable.map(Traversable.scala:104)
        at mimir.lenses.MissingValueLens$.create(MissingValueLens.scala:95)
        at mimir.lenses.LensManager$$anonfun$1.apply(LensManager.scala:17)
        at mimir.lenses.LensManager$$anonfun$1.apply(LensManager.scala:17)
        at mimir.lenses.LensManager.create(LensManager.scala:49)
        at mimir.Database.update(Database.scala:304)
        at mimir.Mimir$.eventLoop(Mimir.scala:118)
        at mimir.Mimir$.main(Mimir.scala:94)
        at mimir.Mimir.main(Mimir.scala)
17:06:25.258 [main] WARN  org.apache.spark.util.Utils - Your hostname, WINDOWS-NFMJ15B resolves to a loopback address: 127.0.0.1, but we couldn't find any external IP address!
17:06:25.258 [main] WARN  org.apache.spark.util.Utils - Set SPARK_LOCAL_IP if you need to bind to another address
apache spark: 2.2.0
```

**Comment by mrb24** [comment]
That is a windows specific hadoop.util.Shell error.  I do not believe that anything we are doing with spark ml uses that, so despite the error, everything should still work and we can ignore the error.  For now I will just turn off the logger for hadoop.util.Shell to prevent the error from printing.

**Comment by mrb24** [comment]
closed by commit: 45e4d515e5b6ee7a2b4f3e7cd3294ad72f52a988
